docs/en/sql-reference/functions/ext-dict-functions.md

```diff
@@ -7,226 +7,188 @@ sidebar_label: Dictionaries
-(
-    key UInt64,
-    start_date Date,
-    end_date Date,
-    value String,
-    value_nullable Nullable(String)
-)
-PRIMARY KEY key
-SOURCE(CLICKHOUSE(HOST 'localhost' PORT tcpPort() TABLE 'range_key_dictionary_source_table'))
-LIFETIME(MIN 1 MAX 1000)
-LAYOUT(RANGE_HASHED())
-RANGE(MIN start_date MAX end_date);
+```response
+'Default return string.'
 ```
 
-Perform the query:
+## dictGetOrNull
 
-``` sql
-SELECT
-    (number, toDate('2019-05-20')),
-    dictHas('range_key_dictionary', number, toDate('2019-05-20')),
-    dictGetOrNull('range_key_dictionary', 'value', number, toDate('2019-05-20')),
-    dictGetOrNull('range_key_dictionary', 'value_nullable', number, toDate('2019-05-20')),
-    dictGetOrNull('range_key_dictionary', ('value', 'value_nullable'), number, toDate('2019-05-20'))
-FROM system.numbers LIMIT 5 FORMAT TabSeparated;
+Retrieves values from a dictionary. If that value cannot be found, returns null. This is different to `dictGet` which throws an exception if the value cannot be found.
+
+**Syntax**
+
+```sql
+dictGetOrNull('dict_name', attr_names, id_expr)
 ```
-Result:
 
-``` text
-(0,'2019-05-20')        0       \N      \N      (NULL,NULL)
-(1,'2019-05-20')        1       First   First   ('First','First')
-(2,'2019-05-20')        1       Second  \N      ('Second',NULL)
-(3,'2019-05-20')        1       Third   Third   ('Third','Third')
-(4,'2019-05-20')        0       \N      \N      (NULL,NULL)
+**Arguments**
+
+- `dict_name` — Name of the dictionary. [String literal](../../sql-reference/syntax.md#syntax-string-literal).
+- `attr_names` — Name of the column of the dictionary, [String literal](../../sql-reference/syntax.md#syntax-string-literal), or tuple of column names, [Tuple](../../sql-reference/data-types/tuple.md)([String literal](../../sql-reference/syntax.md#syntax-string-literal)).
+- `id_expr` — Key value. [Expression](../../sql-reference/syntax.md#syntax-expressions) returning dictionary key-type value or [Tuple](../../sql-reference/data-types/tuple.md)-type value depending on the dictionary configuration.
+
+**Returned value**
+
+If ClickHouse parses the attribute successfully in the [attribute’s data type](../../sql-reference/dictionaries/index.md#dictionary-key-and-fields#ext_dict_structure-attributes), this function returns the value of the dictionary attribute that corresponds to `id_expr`.
+
+If there is no the key, corresponding to `id_expr`, in the dictionary, then this fuction returns `null`.
+
+**Implementation details**
+
+None.
+
+**Example**
+
+Query:
+
+```sql
+CREATE TABLE source_table
+(
+    id UInt64,
+    value String
+) ENGINE = MergeTree
+ORDER BY id;
+
+INSERT INTO source_table VALUES (1, 'First'), (2, 'Second');
+
+CREATE DICTIONARY id_value_dictionary
+(
+    id UInt64,
+    value String
+)
+PRIMARY KEY id
+SOURCE(CLICKHOUSE(TABLE 'source_table'))
+LIFETIME(MIN  0 MAX  1000)
+LAYOUT(FLAT());
+
+SELECT dictGetOrNull('id_value_dictionary', 'missing_attribute', toUInt64(1)) AS value;
```

Comment:
It will throw an exception `Code: 36. DB::Exception: No such attribute 'missing_attribute': While processing dictGetOrNull('id_value_dictionary', 'missing_attribute', toUInt64(1)) AS value. (BAD_ARGUMENTS)`. Please, test that your examples works.

Functions `OrNull/OrDefault` returns default value/null when the requested key doesn't exist, not when

Reply:
Suggestions are welcome.

![image](https://github.com/ClickHouse/ClickHouse/assets/9611008/5aa633ab-07b3-4061-ba6e-651daafc3381)

Reply:
```suggestion
SELECT dictGetOrNull('id_value_dictionary', 'value', toUInt64(3)) AS value;
```